sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py

```diff
@@ -759,7 +759,7 @@ def _to_generated(self):
         )
 
     @classmethod
-    def _from_generated(cls, synonym_map):
+    def _from_generated(cls, synonym_map) -> "SynonymMap":
```

Comment:
Seems like this should be `Optional["SynonymMap"]` since we have the `return None` below?